### PR TITLE
Introduce `dune ocaml` command group

### PR DESCRIPTION
**Pull request by rgrinberg**
This is based on some patches to:
https://github.com/dbuenzli/cmdliner/issues/24#issuecomment-723720963

The work is still ongoing and I plan to improev things further, but we can
already start using it.

This PR uses the new API to introduce an `ocaml` subcommand that includes utop,
top, and merlin. The old commands remain for backwards compatibility.

### CODE REVIEW

**Comment by rgrinberg** [comment]
@jeremiedimino we ended up assigning this to you, since it was your idea to add this group.

File: bin/import.ml
Comment (jeremiedimino) [inline bin/import.ml:32]: This name is not super clear to me. I feel like if we were writing this code from scratch, we'd write `Term t` for each command definition. And so, this function only exist because of the way the code has evolved over the years.

In general, it is better to keep the code looking the same as what we would write from scratch. So here we should push this `Term ...` in all command definitions.

**Review by ghost — CHANGES_REQUESTED** [review]
We should probably use the new commands in the tests (i.e. `dune ocaml top` instead of `dune top`). Just to make sure it's tested.

File: bin/import.ml
Comment (rgrinberg) [reply bin/import.ml:32]: Yes, that's a good point, but I intentionally wrote it this way to minimize the diff. My reasoning is that Daniel might want to tweak the api, so it's better to delay changing our entire codebase until the api is settled.

Comment (jeremiedimino) [reply bin/import.ml:32]: Makes sense. We should also wait until the design is settled on the cmdliner side before merging then, to avoid having the behaviour changing (I remember that they were some discussions around the presentation of the help). Could you put WIP in the title?

Comment (rgrinberg) [reply bin/import.ml:32]: @jeremiedimino i think we should reconsider merging this PR. It's currently blocking working on dune rpc, and it would also be useful in other dune 3 related clean ups. Daniel already what kind of design he wants, and this PR implements exactly that. I'm just waiting for code review of the implementation at this point. Whatever changes to the api he makes are going to be trivial for me to adopt anyway. 

So let's just merge this now and tweak this again once upstream is updated.

Comment (jeremiedimino) [reply bin/import.ml:32]: SGTM!

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**
